pkg/skaffold/docker/image.go

```diff
@@ -218,6 +218,26 @@ func (l *localDaemon) Push(ctx context.Context, out io.Writer, ref string) (stri
 		return "", errors.Wrapf(err, "getting auth config for %s", ref)
 	}
 
+	// Check if the local image has a digest from a previous push
+	// If the remote image has the same digest, there's no need
+	// to push.
+	image, _, err := l.apiClient.ImageInspectWithRaw(ctx, ref)
+	if err != nil {
+		return "", errors.Wrapf(err, "inspecting image %s", ref)
+	}
+
+	if len(image.RepoDigests) == 1 {
```

Comment:
could it not be the case that we have multiple `RepoDigest`s, but one of them matches a previous push from skaffold? do we not want to iterate through these here?

Reply:
I have no idea why There's a list here. I decided to not go crazy and only handle the case I can comprehend :-)

Reply:
lol. fair enough